llvm/include/llvm/ADT/APInt.h

```diff
@@ -2294,6 +2294,12 @@ LLVM_ABI APInt mulhs(const APInt &C1, const APInt &C2);
 /// Returns the high N bits of the multiplication result.
 LLVM_ABI APInt mulhu(const APInt &C1, const APInt &C2);
 
+/// Performs (2*N)-bit multiplication on sign-extended operands.
+LLVM_ABI APInt mulxs(const APInt &C1, const APInt &C2);
+
+/// Performs (2*N)-bit multiplication on zero-extended operands.
+LLVM_ABI APInt mulxu(const APInt &C1, const APInt &C2);
```

Comment:
These are called `mulsi_` and `mului_extended` in MLIR, FYI: https://mlir.llvm.org/docs/Dialects/ArithOps/#arithmulsi_extended-arithmulsiextendedop

I think this name is more descriptive -- I expect this not to be used all over the codebase, so I wouldn't optimize for conciseness.

Reply:
I wasn't aware those existed in MLIR.
With that in mind, it definitely makes more sense to use those names.